src/controllers/up-git-dependencies.js

```diff
@@ -0,0 +1,143 @@
+/*!
+ * V4Fire cli
+ * https://github.com/V4Fire/cli
+ *
+ * Released under the MIT license
+ * https://github.com/V4Fire/cli/blob/master/LICENSE
+ */
+
+const
+	util = require('util'),
+	exec = util.promisify(require('child_process').exec);
+
+const
+	{Controller} = require('../core/controller');
+
+require('@v4fire/core');
+
+ class UpGitDependencies extends Controller {
+	/**
+	 * List of Git project dependencies
+	 */
+	gitDependencies = [];
+
+	/**
+	 * Path to the project lockfile
+	 */
+	lockFilePath = this.vfs.resolve('yarn.lock');
+
+	/**
+	 * Path to the project `package.json` file
+	 */
+	packageJSONPath = this.vfs.resolve('package.json');
+
+	/**
+	 * Updates all Git dependencies
+	 *
+	 * @returns {!Promise<void>}
+	 */
+	async run() {
+		this.extractGitDependencies();
+		this.removeGitDependenciesFromLockFile();
+
+		await this.installDependencies();
+	}
+
+	/**
+	 * Extracts all Git dependencies from the project `package.json`
+	 *
+	 * @returns {Array}
+	 */
+	extractGitDependencies() {
+		const {packageJSON} = this;
+
+		for (const key in packageJSON) {
+			if (key in packageJSON) {
+				const value = packageJSON[key];
+
+				if (/^((git)|(https))/.exec(value) != null) {
+					this.gitDependencies.push(key);
+				}
+			}
+		}
+	}
+
+	/**
+	 * Returns content of package.json
+	 *
+	 * @returns {Object}
+	 */
+	get packageJSON() {
+		let
+			packageJSON;
+
+		try {
+			packageJSON = this.vfs.readFile(this.packageJSONPath);
+
+		} catch {
+			throw new Error('An error occurred when reading package.json');
+		}
+
+		const parsedPackageJSON = JSON.parse(packageJSON);
+
+		return {
+			...parsedPackageJSON.dependencies,
+			...parsedPackageJSON.devDependencies,
+			...parsedPackageJSON.optionalDependencies
+		};
+	}
+
+	/**
+	 * Returns content of lockfile
+	 *
+	 * @returns {Object}
```

Comment:
```suggestion
	 * @returns {!Object}
```

А нет более адекватного интерфейса?